### ISSUE TITLE
provider/vsphere: failing acceptance tests because of hardcoded disk id

### ISSUE THREAD

**Issue by thetuxkeeper** [hashicorp/terraform#6964]
Hi, the acceptance tests fail because the disk id depends on the environment/template.
@dkalleg : could be related to your change to the disk stuff
```
--- FAIL: TestAccVSphereVirtualMachine_mac_address (89.25s)
        testing.go:239: Step 0 error: Check failed: Check 9/9 error: vsphere_virtual_machine.mac_address: Attribute 'disk.2166312600.template' expected "terraform-test-folder/centos-terraform-template", got ""
FAIL
exit status 1
```

```
STATE:

vsphere_virtual_machine.mac_address:
  ID = terraform-mac-address
  cluster = dc1.boerse-go.de
  datacenter = dc1
  disk.# = 1
  disk.1377756793.bootable = false
  disk.1377756793.controller_type = scsi
  disk.1377756793.datastore = DS1
  disk.1377756793.iops = 0
  disk.1377756793.keep_on_remove = false
  disk.1377756793.key = 0
  disk.1377756793.name = 
  disk.1377756793.size = 0
  disk.1377756793.template = terraform-test-folder/centos-terraform-template
  disk.1377756793.type = eager_zeroed
  disk.1377756793.uuid = 
  disk.1377756793.vmdk = 
  domain = vsphere.local
  linked_clone = false
  memory = 4096
  memory_reservation = 0
  name = terraform-mac-address
  network_interface.# = 1
  network_interface.0.adapter_type = 
  network_interface.0.ip_address = 
  network_interface.0.ipv4_address = 10.30.8.52
  network_interface.0.ipv4_gateway = 10.30.8.1
  network_interface.0.ipv4_prefix_length = 23
  network_interface.0.ipv6_address = fe80::250:56ff:fe3f:ffff
  network_interface.0.ipv6_gateway = 
  network_interface.0.ipv6_prefix_length = 64
  network_interface.0.label = cld_tst1_access
  network_interface.0.mac_address = 00:50:56:3f:ff:ff
  network_interface.0.subnet_mask = 
  resource_pool = dc1.boerse-go.de/Resources/test_server
  skip_customization = false
  time_zone = Etc/UTC
  vcpu = 2
```
cc @chrislovecnm

**Comment by chrislovecnm** [comment]
I have a PR that takes car of this ;)

**Comment by chrislovecnm** [comment]
https://github.com/hashicorp/terraform/pull/6893

**Comment by chrislovecnm** [comment]
Also asking how we can fix this as well https://github.com/hashicorp/terraform/issues/6959

**Comment by dkalleg** [comment]
@thetuxkeeper This is definitely due to my PR, you don't have to sugar coat it :) .  Things were passing on my end when I submitted but I didn't realize at the time how the hash will change depending on the different user args.  This would have been caught before the merge if the tests were run anywhere else.  @chrislovecnm did you mention you were looking into automated testing hooks?  This is a prime example of why they would be super useful.

I've got some work I can't move aside that'll bring me into mid-next week.  After that I can start working on a fix for this.  Anyone else with bandwidth should feel free! :)

**Comment by chrislovecnm** [comment]
@dkalleg my PR backs out your disk changes, and I have the other question in to help us fix this.  Yes automated testing is next on my plate after my big DRS PR.

**Comment by thetuxkeeper** [comment]
@dkalleg: hehe :) just wanted to let you know. I'm just back from vacation and had a quick look into the commit log after I saw the tests failing (was implementing a smaller feature)
@chrislovecnm: better/easier testing, best automated, is really important! I'll be on slack tomorrow again, was busy today.

**Comment by chrislovecnm** [comment]
To let you guys know.  You need to merge my testing changes sooner than later.  That PR will be fun to merge into.

**Comment by stack72** [comment]
Closed by #6893
